### PR TITLE
Make np.divide and np.remainder code more similar

### PR DESCRIPTION
**Pull request by eric-wieser**
Split out of #5619. Some simple cleanup to make adding `divrem` easier.

### CODE REVIEW

**Comment by stuartarchibald** [comment]
Thanks for the patch. I've queued it for review.

File: numba/np/npyfuncs.py [hunk 1]
```diff
@@ -161,10 +162,9 @@ def np_int_srem_impl(context, builder, sig, args):
     _check_arity_and_homogeneity(sig, args, 2)
 
     num, den = args
-    ty = sig.args[0] # any arg type will do, homogeneous
-    lty = num.type
+    lltype = num.type
 
-    ZERO = context.get_constant(ty, 0)
+    ZERO = lc.Constant.int(lltype, 0)
```
Comment (stuartarchibald) [inline numba/np/npyfuncs.py:167]: There's a general preference in Numba to use the `context` opposed to `llvmpy`.

Comment (eric-wieser) [reply numba/np/npyfuncs.py:167]: So you'd want me to make the change in the reverse direction?

Comment (eric-wieser) [reply numba/np/npyfuncs.py:167]: (by changeing line 194 below instead?)

Comment (eric-wieser) [reply numba/np/npyfuncs.py:167]: Also, does numba provide any helper to convert between the numba and llvm types?

Comment (stuartarchibald) [reply numba/np/npyfuncs.py:167]: Yes please, the reverse direction change is appropriate. The context has loads of helper functions on it, to find the llvm equivalent of a numba type use `context.get_value_type`. e.g. some bits and pieces from breaking in the `ii->i` loop which hits `np_int_srem_impl`:
```
(Pdb) sig.args
(int32, int32)
(Pdb) argty = context.get_value_type(sig.args[0])
(Pdb) p argty
<<class 'llvmlite.ir.types.IntType'> i32>
(Pdb) ZERO = context.get_constant_generic(builder, sig.args[0], 0)
(Pdb) p ZERO
<ir.Constant type='i32' value=0>
```

Comment (stuartarchibald) [reply numba/np/npyfuncs.py:167]: hmmm, this might not work here, just need to test something.

Comment (stuartarchibald) [reply numba/np/npyfuncs.py:167]: checked, it's ok.

Comment (eric-wieser) [reply numba/np/npyfuncs.py:167]: To be clear, I'm asking if the type can be obtained in the other direction

Comment (stuartarchibald) [reply numba/np/npyfuncs.py]: Don't think so, the map would be one llvm type -> many Numba types?
```
(Pdb) context.get_value_type(types.int8)
<<class 'llvmlite.ir.types.IntType'> i8>
(Pdb) context.get_value_type(types.uint8)
<<class 'llvmlite.ir.types.IntType'> i8>
(Pdb) context.get_value_type(types.char)
<<class 'llvmlite.ir.types.IntType'> i8>
```

Comment (eric-wieser) [reply numba/np/npyfuncs.py]: Good point. Sounds like I need to modify #5619 to construct the signature too then.

Comment (eric-wieser) [reply numba/np/npyfuncs.py]: And by that, I'm referring to these lines: https://github.com/numba/numba/pull/5619/files#diff-6fb12019cea9375115bc5f3b7a71fe3dR277

**Pull request merged**